### PR TITLE
 Implement pagination for search/filtering results, fixes #5

### LINKED ISSUE #5: Implement pagination
Declaration list pages should be paginated.

### PR DESCRIPTION
**Pull request by excieve**
- Implements an ES-minded subclass of Django's Paginator
- Adds a contextual page numbering
- Adds a helper Jinja2 global for updating querystrings
- Adds a pagination widget to the results template
- Corresponding view changes

### CODE REVIEW

**Comment by dchaplinsky** [comment]
Нихуево!

File: declarations_site/catalog/paginator.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+from django.core.paginator import Paginator, Page
+from django.utils import six
+from django.conf import settings
+
+from elasticsearch_dsl.result import Response
+
+
+class ElasticPaginator(Paginator):
```
Comment (dchaplinsky) [inline declarations_site/catalog/paginator.py:8]: Wanna to make a PR to elasticsearch_dsl contribs?

Comment (excieve) [inline declarations_site/catalog/paginator.py:8]: They don't have contribs and I doubt they would want that. Maybe some Django wrapper like this instead:
https://github.com/Sparrho/bungiesearch

**Comment by excieve** [comment]
@dchaplinsky, feel free to merge if you're happy with it.

**Pull request merged**